libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_DNS.c

```diff
@@ -1112,321 +1109,341 @@ uint16_t usType = 0U;
-				if( x == 0U )
+				#if( ipconfigUSE_LLMNR == 1 )
 				{
-					pcRequestedName = ( char * ) pucByte;
+					if( x == 0U )
+					{
+						pcRequestedName = ( char * ) pucByte;
+					}
 				}
-			}
-			#endif
+				#endif
 
 #if( ipconfigUSE_DNS_CACHE == 1 ) || ( ipconfigDNS_USE_CALLBACKS == 1 )
-			if( x == 0U )
-			{
-				uxResult = prvReadNameField( pucByte,
-											 uxSourceBytesRemaining,
-											 pcName,
-											 sizeof( pcName ) );
-
-				/* Check for a malformed response. */
-				if( uxResult == 0U )
+				if( x == 0U )
 				{
-					return dnsPARSE_ERROR;
+					uxResult = prvReadNameField( pucByte,
+												 uxSourceBytesRemaining,
+												 pcName,
+												 sizeof( pcName ) );
+
+					/* Check for a malformed response. */
+					if( uxResult == 0U )
+					{
+						xReturn = pdFALSE;
+						break;
+					}
+					uxBytesRead += uxResult;
+					pucByte = &( pucByte[ uxResult ] );
+					uxSourceBytesRemaining -= uxResult;
 				}
-				uxBytesRead += uxResult;
-				pucByte = &( pucByte[ uxResult ] );
-				uxSourceBytesRemaining -= uxResult;
-			}
-			else
+				else
 #endif /* ipconfigUSE_DNS_CACHE || ipconfigDNS_USE_CALLBACKS */
-			{
-				/* Skip the variable length pcName field. */
-				uxResult = prvSkipNameField( pucByte,
-											 uxSourceBytesRemaining );
-
-				/* Check for a malformed response. */
-				if( uxResult == 0U )
 				{
-					return dnsPARSE_ERROR;
+					/* Skip the variable length pcName field. */
+					uxResult = prvSkipNameField( pucByte,
+												 uxSourceBytesRemaining );
+
+					/* Check for a malformed response. */
+					if( uxResult == 0U )
+					{
+						xReturn = pdFALSE;
+						break;
+					}
+					uxBytesRead += uxResult;
+					pucByte = &( pucByte[ uxResult ] );
+					uxSourceBytesRemaining -= uxResult;
 				}
-				uxBytesRead += uxResult;
-				pucByte = &( pucByte[ uxResult ] );
-				uxSourceBytesRemaining -= uxResult;
-			}
 
-			/* Check the remaining buffer size. */
-			if( uxSourceBytesRemaining >= sizeof( uint32_t ) )
-			{
-				#if( ipconfigUSE_LLMNR == 1 )
+				/* Check the remaining buffer size. */
+				if( uxSourceBytesRemaining >= sizeof( uint32_t ) )
 				{
-					/* usChar2u16 returns value in host endianness. */
-					usType = usChar2u16( pucByte );
-					usClass = usChar2u16( &( pucByte[ 2 ] ) );
-				}
-				#endif /* ipconfigUSE_LLMNR */
+					#if( ipconfigUSE_LLMNR == 1 )
+					{
+						/* usChar2u16 returns value in host endianness. */
+						usType = usChar2u16( pucByte );
+						usClass = usChar2u16( &( pucByte[ 2 ] ) );
+					}
+					#endif /* ipconfigUSE_LLMNR */
 
-				/* Skip the type and class fields. */
-				pucByte = &( pucByte[ sizeof( uint32_t ) ] );
-				uxSourceBytesRemaining -= sizeof( uint32_t );
+					/* Skip the type and class fields. */
+					pucByte = &( pucByte[ sizeof( uint32_t ) ] );
+					uxSourceBytesRemaining -= sizeof( uint32_t );
+				}
+				else
+				{
+					xReturn = pdFALSE;
+					break;
+				}
 			}
-			else
+
+			if( xReturn == pdFAIL )
```

Comment:
pdFALSE maybe? (Mixing pdFALSE/pdTRUE with pdFAIL/pdPASS may be confusing.)

Reply:
Oops! Thanks for spotting that :)